lib/Dialect/ESI/runtime/cpp/lib/Accelerator.cpp

```diff
@@ -45,6 +54,74 @@ services::Service *AcceleratorConnection::getService(Service::Type svcType,
   return cacheEntry.get();
 }
 
+/// Get the path to the currently running executable.
+static std::filesystem::path getExePath() {
+#ifdef __linux__
+  char result[PATH_MAX];
+  ssize_t count = readlink("/proc/self/exe", result, PATH_MAX);
+  if (count == -1)
+    throw runtime_error("Could not get executable path");
+  return std::filesystem::path(std::string(result, count));
+#elif _WIN32
+  // TODO: this.
+#else
+#eror "Unsupported platform"
+#endif
+}
+
+/// Get the path to the currently running shared library.
+static std::filesystem::path getLibPath() {
+#ifdef __linux__
+  Dl_info dl_info;
+  dladdr((void *)getLibPath, &dl_info);
+  return std::filesystem::path(std::string(dl_info.dli_fname));
+#elif _WIN32
+  // TODO: this.
+#else
+#eror "Unsupported platform"
+#endif
+}
+
+static void loadDynamic(string backend) {
```

Comment:
> 1. `const std::string&`
>    my eyes always go oogley reading files with `using namespace std`;.

Yeah, using `namespace std` is a bad habit I got into when first learning C++ in high school.

Can't use const since the string gets modified.

Reply:
> 3\. I think there should be two functions; one as is (which tries to infer the backend path), and one which takes an absolute path to a backend. I'd imagine there should also be some option for users of ESI to manually provide the path to a backend, instead of relying on relative directory and backend name heuristics.

I think it's fair to use LD_PRELOAD for other non-standard path plugins.